cmd/keys_rotate.go

```diff
@@ -0,0 +1,230 @@
+                    keyname = x[:-4]  # strip off .pub
+                    print("current root keyname:", keyname)
+                    return keyname, key_ids[0]
+    print('could not find root key name')
+    sys.exit(1)
+
+with TemporaryDirectory() as tempdir:
+    os.chdir(tempdir)
+    os.mkdir('tuf')
+    creds_file = '/tmp/creds.tgz'
+    cmd('tar', 'xf', creds_file, cwd='./tuf')
+    cmd('garage-sign', 'root', 'pull', '--repo', './tufrepo')
+    old_keyname, old_keyid = find_current_root('./tuf/tufrepo')
+    keyname = 'offline-root-' + datetime.datetime.now().isoformat()
+    cmd('garage-sign', 'key', 'generate', '--repo', './tufrepo',
+    '--type', 'rsa', '--name', keyname)
+    cmd('garage-sign', 'root', 'key', 'add', '--repo', './tufrepo',
+    '--key-name', keyname)
+    cmd('garage-sign', 'root', 'key', 'remove', '--repo', './tufrepo',
+    '--key-id', old_keyid, '--key-name', old_keyname)
+    cmd('garage-sign', 'root', 'sign', '--repo', './tufrepo',
+    '--key-name', keyname, '--key-name', old_keyname)
+    cmd('tar', 'czf', creds_file, 'tufrepo', cwd='./tuf')
+    cmd('garage-sign', 'root', 'push', '--repo', './tufrepo')
+`
+	fmt.Println("Pulling aktualizr image...")
+	if err := pullContainer(aktualizrImageName); err != nil {
+		exitf("failed to pull image, %q, %s", aktualizrImageName, err)
+	}
+	scriptPath, err := loadScript(rotateCmd)
+	if err != nil {
+		exitf("failed to load script, %s", err)
+	}
+	defer os.Remove(scriptPath) // clean up
+	if err := copyFile(credentialsPath, credentialsBackupPath); err != nil {
+		exitf("failed to backup offline credentials, %s", err)
+	}
+	fmt.Printf("Original credentials --> %q\n", credentialsBackupPath)
+	fmt.Println("Rotating offline credentials...")
+	if err := runRotationScript(aktualizrImageName, scriptPath, credentialsPath); err != nil {
+		exitf("failed key rotation %s", err)
+	}
+	fmt.Printf("Updated credentials --> %q\n", credentialsPath)
+}
+
+func exitf(format string, args ...interface{}) {
+	errFormat := fmt.Sprintf("Error: %s\n", format)
+	fmt.Printf(errFormat, args...)
+	os.Exit(1)
+}
+
+func loadScript(script string) (string, error) {
+	content := []byte(script)
+	tmpFile, err := ioutil.TempFile("/tmp", "*tmpScript")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary script: %w", err)
+	}
+	scriptPath := tmpFile.Name()
+	if _, err := tmpFile.Write(content); err != nil {
+		return "", fmt.Errorf("failed to write temporary script: %w", err)
+	}
+	if err := tmpFile.Close(); err != nil {
+		return "", fmt.Errorf("failed to save temporary script: %w", err)
+	}
+	if err := os.Chmod(scriptPath, 0777); err != nil {
+		return "", fmt.Errorf("failed to make script executable: %w", err)
+	}
+	return scriptPath, nil
+}
+
+func runRotationScript(imageName string, sourcePath string, credentialsPath string) error {
+	targetPath := "/tmp/tmp.py"
+	args := []string{
+		// base args
+		"run", "--rm",
+		// env args
+		"--env", "PYTHONUNBUFFERED=1",
```

Comment:
ah - i bet that was driving you mad.

Reply:
it was 😄

Reply:
quick bit of history, I initially tried using the tuft/client (dead) code  we've been talking about and it didn't change output at all (or so I thought at the time).  After switching back and trying with a Pipeline -- same issue, so I then just ran the command directly and noticed that docker wasn't outputting til the end.  After switching that environment variable - it was fixed.  Not with "CombinedOutput" though ( which I tried when doing this streaming version) -- hence the use again of this code. 

Took quite a tour to get here.  It was fun.